client/src/app/payroll/payroll-list/state/payroll-list.service.ts

```diff
@@ -0,0 +1,45 @@
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+import { PayrollList } from './payroll-list.model';
+import { PayrollListStore } from './payroll-list.store';
+
+import { ToastrService } from 'ngx-toastr';
+import { EnvService } from 'src/app/shared/env.service';
+import { ListResponse, mapListResponseToData } from 'src/app/shared/list-response';
+
+@Injectable({ providedIn: 'root' })
+export class PayrollListService {
+  readonly apiRootUrl: string;
+
+  constructor(
+    envService: EnvService,
```

Comment:
Not something you need to (or even should) change, but there's a cute way to get a single value from an injected service/object.

```typescript
constructor(
  { apiRootUrl }: EnvService,
  private readonly http: HttpClient,
  ...
) {
  this.apiRootUrl = apiRootUrl;
}
```

[Playground Link](https://www.staging-typescript.org/play?#code/MYGwhgzhAECC0G8BQ1XQHZgLYFMBc0EALgE4CW6A5tALzQDk9A3EgL5JKiQwBCiKaTLloNmHNNGAB7dMRIBXYESkkAFAImoEGbDmisCsDdACU-TWiIALMhAB0QvXUcsJ7VkA)